### PR TITLE
tools: Tidy up some of the filepath-handling code

### PR DESCRIPTION
**Pull request by lurch**

### CODE REVIEW

**Comment by dpgeorge** [comment]
Thanks for the PR!

Are any of these needed, in the sense that they fix issues on Windows/Mac/other?

Note that `uf2conv.py` is sourced from an upstream repo so best to leave that alone (to make it easier to pull in changes to that file if needed).

File: tools/gen-cpydiff.py [hunk 1]
```diff
@@ -197,11 +198,8 @@ def gen_rst(results):
     """ creates restructured text documents to display tests """
 
     # make sure the destination directory exists
-    try:
+    if not os.path.isdir(DOCPATH):
         os.mkdir(DOCPATH)
-    except OSError as e:
-        if e.args[0] != errno.EEXIST and e.args[0] != errno.EISDIR:
-            raise
```
Comment (dpgeorge) [inline tools/gen-cpydiff.py:204]: This is definitely an improvement.

File: tools/pyboard.py [hunk 2]
```diff
@@ -482,13 +482,11 @@ def fname_remote(src):
         return src
 
     def fname_cp_dest(src, dest):
-        src = src.rsplit("/", 1)[-1]
+        src = os.path.basename(src)
```
Comment (dpgeorge) [inline tools/pyboard.py:485]: I guess this fixes a bug where it wouldn't work on Windows using `\` as separators?

File: tools/pyboard.py [hunk 3]
```diff
@@ -482,13 +482,11 @@ def fname_remote(src):
...
         if dest is None or dest == "":
             dest = src
-        elif dest == ".":
-            dest = "./" + src
-        elif dest.endswith("/"):
-            dest += src
+        elif dest == "." or dest.endswith("/"):
```
Comment (stinos) [inline tools/pyboard.py:488]: Is this intended, seeing you're using os.path functions anywhere improving cross-platform behaviour?

**Comment by lurch** [comment]
> Are any of these needed, in the sense that they fix issues on Windows/Mac/other?

Nope, strictly speaking this isn't "needed", and I'm not aware of any bugs it fixes (and I've not tested it on Windows or Mac).
I was just "inspired" by https://github.com/micropython/micropython/commit/448319a7450714efd72ae527e184a247a82ee39d and thought I'd see if there were any other scripts that needed "tidying up".

> Note that uf2conv.py is sourced from an upstream repo so best to leave that alone (to make it easier to pull in changes to that file if needed).

I _did_ consider that, but I saw from it's [file history](https://github.com/micropython/micropython/commits/master/tools/uf2conv.py) that it had already been edited in this repo :wink: 

Feel free to reject this PR if you feel it's inappropriate.

File: tools/pyboard.py (hunk 3 above)
Comment (lurch) [reply tools/pyboard.py:488]: Dunno, I was trying to do code-tidyups without changing existing behaviour.

**Comment by jimmo** [comment]
@dpgeorge I have rebased this -- see the two commits here: https://github.com/jimmo/micropython/commits/lurch-filepath_tidyup

> I _did_ consider that, but I saw from its [file history](https://github.com/micropython/micropython/commits/master/tools/uf2conv.py) that it had already been edited in this repo 😉

The only changes are to reformat to black, and then to update to a later version. The idea is that when we update, we just copy the upstream exactly, then run black. I have removed this file from the PR.

What's left is just the os.path changes to `gen-cpydiff.py`.

`make-frozen.py` was removed from the main repo since this PR was created. Also gendoc.py is unused, so I have just made a commit to remove it instead. The change to pyboard.py had already been made in a similar way as part of other recent changes.

**Comment by dpgeorge** [comment]
New version of this PR is at #9967.

**Comment by dpgeorge** [comment]
The changes to `gen-cpydiff.py` were merged in 02ad71468f1d52f9521bf89dd8b9594a4d437e51

**Pull request closed without merging**